DEVELOPMENT.md

```diff
@@ -0,0 +1,107 @@
+
+The goal of this project is to make it as easy as possible to get up and running with open source FPGA tools. A secondary goal is to provide the same set of tools and features on all platforms, wherever possible.
+
+To achieve these goals, some compromises have been made:
+ * Static linking is used in almost all cases - removing dependencies on external shared libraries means this package should be all you need to run the tools. The major downside of this approach is that some tools are not designed to be built statically, which can make it harder or near-impossible to include them.
+ * The goal is not to rely on any absolute install prefixes - tools must be able to find any files they need at runtime by using relative paths wherever possible. This also enables multiple versions of the tools to be installed without conflicting.
+ * The package is provided as a simple compressed archive - no package managers are used. This makes installation simple, but the downside is that **before accepting a contribution adding a new tool, we need to consider how much larger it makes the overall package**.
+
+# Development
+
+## Build Prerequisites
+At present, all of the builds are run on their native platform (no cross-compiling).
+
+**Windows**
+
+ 1. Install MSYS2 - install options:
+    * Use [chocolately](https://chocolatey.org/) and run `choco install msys2`
+    * Install using instructions on [the MSYS2 website](https://www.msys2.org/)
+ 2. Run the MinGW64 bash environment `mingw64.exe` (not the MSYS2 or MinGW32 environments).
+
+**macOS**
+
+1. Install [Xcode 11.4.1](https://developer.apple.com/services-account/download?path=/Developer_Tools/Xcode_11.4.1/Xcode_11.4.1.xip) to `/Applications/Xcode_11.4.1.app` (other versions might work but this is untested and unsupported)
+
+*Note that this requires macOS 10.15.2+ as a build-time dependency. The resulting builds should work on versions as old as macOS 10.10*
+
+2. Install [Homebrew](https://brew.sh/)
+
+**Linux**
+ 1. Create an Ubuntu 20.04 environment:
+    * The `ubuntu:20.04` docker container is a good option and this is the approach the CI build uses
+    * A virtual machine or even WSL on Windows will work too
+    * Running on bare metal is of course fine too, but other versions of Ubuntu are untested and unsupported.
+
+*Note that Ubuntu 20.04 is only a build-time dependency. The resulting builds are intended to run on any Linux distro*
+
+ 2. Make sure the `sudo` package is installed (it is not installed by default in the docker image)
+
+## Running a Build
+
+**WARNING**: The scripts will attempt to automatically install build dependencies by default! If you don't want this you can disable it first with `export INSTALL_DEPS=0`.
+
+The details of the required dependencies will not be documented here, so it is recommended to just let the scripts handle it - you can see what will be installed in `scripts/install_dependencies.sh`
+
+Build:
+
+```
+bash build.sh <arch>
+```
+
+Clean:
```

Comment:
It would be good to add, I think, when a "clean" is recommended. Does it always assume currently to start from a clean build tree? For example I just tried to do
```
./build_bba.sh linux_x86_64
COMPILE_NEXTPNR_ECP5=1 ./build.sh linux_x86_64
```
And I get various "mv: … Directory not empty" errors in the "Bundle Python" step. This is likely to be from leftovers from a previous build.

Reply:
I've added a note about this. It would probably be a "nice to have" feature if it was less prone to breaking when the tree isn't clean, but it hasn't been something I prioritised.